python/plugins/processing/algs/qgis/DefineProjection.py

```diff
@@ -77,16 +85,19 @@ def processAlgorithm(self, parameters, context, feedback):
         if dsPath.lower().endswith('.shp'):
             dsPath = dsPath[:-4]
 
-            wkt = crs.toWkt()
+            wkt = crs.toWkt(QgsCoordinateReferenceSystem.WKT1_ESRI)
             with open(dsPath + '.prj', 'w') as f:
                 f.write(wkt)
 
             qpjFile = dsPath + '.qpj'
             if os.path.exists(qpjFile):
-                with open(qpjFile, 'w') as f:
-                    f.write(wkt)
+                if QgsProjUtils.projVersionMajor() < 6:
+                    with open(qpjFile, 'w') as f:
+                        f.write(wkt)
```

Comment:
@nyalldawson , are you sure WKT1_ESRI is the format used for the qpj?

Reply:
It doesn't matter. For proj < 6 the format argument is ignored, and for >= 6 the qpj file is deleted, not written to

Reply:
Good, good.